### ISSUE TITLE
Large git clone timeouts even if it progresses

### ISSUE THREAD

**Issue by mvorisek** [composer/composer#9936]
Cloning large packages using git (10+ GB, 100k+ files) is slow and the 300s timeout is not enough.

Shallow clone #3449 can help, #5384 still does not solves this issue completely. However I understand shallow clone might not be always the solution.

## This issue is a feature request to NOT timeout if the git report progress.

Example:

```
remote: Enumerating objects: 1910, done.
remote: Counting objects: 100% (1910/1910), done.
remote: Compressing objects: 100% (122/122), done.
Receiving objects:  1% (80227/7686857), 18.77 MiB | 4.66 MiB/s
```

and when the `Receiving objects` count is increased, reset the process timeout. The git progress should be reported also to the user. Currently, there is not progress visible at all.

**Comment by Seldaek** [comment]
If the timeout is not enough, maybe increase the timeout? This is a default for most use cases. If you clone a huge repo then maybe you need to go beyond the defaults.

**Comment by mvorisek** [comment]
Yes, I increased it. But composer should reset the timeout when there is an expected progress (like `Receiving objects` count is increased) and this progress should be also presented to the user.

**Comment by Seldaek** [comment]
Feel free to work on a PR if it can be done without too much code why not but otherwise IMO not worth it.

**Comment by stof** [comment]
symfony/process has 2 different timeout concepts: the timeout (used by composer) and the idle timeout (not used by composer today).

Implementing this would mean allowing to configure both in composer (with a much longer timeout value)

**Comment by stof** [comment]
note that the idle timeout is available since Symfony 2.4 so it is available to composer even without the bump of the min version that is planned for 2.2.
